libraries/chain/apply_context.cpp

```diff
@@ -290,6 +290,10 @@ void apply_context::execute_inline( action&& a ) {
 
    // No need to check authorization if replaying irreversible blocks or contract is privileged
    if( !control.skip_auth_check() && !privileged ) {
+      const auto& chain_config = control.get_global_properties().configuration;
+      EOS_ASSERT( a.data.size() < std::min(chain_config.max_inline_action_size, control.get_max_nonprivileged_inline_action_size()),
+                  inline_action_too_big_nonprivileged,
+                  "inline action too big for nonprivileged account ${account}", ("account", a.account));
```

Comment:
I think this needs to be in an `if (!privileged && control.is_producing_block())`

Reply:
Done.